host/sgx/internal_u_wrapper.c

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+
+#include <openenclave/edger8r/host.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/raise.h>
+
+/* Override oe_call_enclave_function() calls with _call_enclave_function(). */
+#define oe_call_enclave_function _call_enclave_function
+
+/* The ocall edge routines will use this function to route ecalls. */
+static oe_result_t _call_enclave_function(
+    oe_enclave_t* enclave,
+    uint32_t function_id,
+    const void* input_buffer,
+    size_t input_buffer_size,
+    void* output_buffer,
+    size_t output_buffer_size,
+    size_t* output_bytes_written)
+{
+    return oe_call_enclave_function_by_table_id(
+        enclave,
+        OE_INTERNAL_ECALL_FUNCTION_TABLE_ID,
+        function_id,
+        input_buffer,
+        input_buffer_size,
+        output_buffer,
+        output_buffer_size,
+        output_bytes_written);
+}
+
+/* Ignore missing edge-routine prototypes. */
```

Comment:
Can we not include internal_u.h which contains the prototypes for edge functions?

Reply:
I added an inclusion but that header does not define certain internal functions (which probably should have been declared in **internal_u.h**).

Reply:
@mikbras edger8r generates prototypes for all functions it defines that the user can call. 
The prototypes for the edger8r infrastructure functions reside in https://github.com/openenclave/openenclave/blob/master/include/openenclave/edger8r/host.h

Can you create a new issue if additional prototypes need to be generated by edger8r?